### ISSUE TITLE
Can't get shortcut keys to work...

### ISSUE THREAD

**Issue by hikerguy** [redmunds/brackets-quick-markup#36]
I've been using Quick Markup for a few weeks now with no problems, but tonight it decided to act up. I made sure I had Quick Markup activated (Ctrl+Shift+M) and tried several key combinations (e.g. Ctrl+p, Ctrl+1, Ctrl+2) that had worked flawlessly before. Now nothing is working.

I have tried the following with no change in the problem:

Closed up Brackets and restarted it.

Uninstalled and installed Quick Markup. No change, so I closed up Brackets and restarted it. Still no change. 

Tried it with Caps Lock both on and off. 

Rebooted my PC for grins. 

I am running Windows 7 and have the latest version of Brackets (went to Help | Check for Updates).

Any idea why this stopped working all of a sudden?  This has been a very useful tool for me and I'd really like to get it working again.

Thanks,

Andy

**Comment by redmunds** [comment]
Start by looking at the dev tools console to see if there are any errors. Paste any errors you see in here.

Note: you can turn on *Debug > Show Errors in Status Bar* so you can see the number of errors displayed and click on indicator to startup dev tools.

Did you make any custom changes to the Quick Markup **data.json** file? If so, make sure there are no syntax errors which would prevent code from reading shortcuts.

Have you overridden any shortcuts using *Debug > Open User Key Map* ?

**Comment by hikerguy** [comment]
Randy,

          Every time I tried to use a key sequence to place tags around a
word (e.g. Ctrl+p, Ctrl+1), the Error counter incremented.  Here's what
it's saying (regardless of the key sequence I use to add a tag):

Uncaught TypeError: Cannot read property 'tagName' of null
handleBlockTag
getEdits
(anonymous function)
handleKey
_keydownHook
_handleKeyEvent

Is there a way to export these error messages to a file? I tried to
highlight, right-click and copy, but Copy wasn't an option. Also, Ctrl+c
wouldn't copy the highlighted text either.

To answer your other questions:

I did not modify data.json (wasn't even aware of that file). I didn't
modify any key mappings (I wasn't aware of that feature either).

Any idea what's going on?

Thanks

Andy

**Comment by redmunds** [comment]
This works for me in latest version of Brackets (1.2) and running on Windows 7. I can also copy text from dev tools console.

This could be a conflict with another extension, so try disabling all other extensions.

If not, then can you post some sample code? I'm using a very simple .html file.

**Comment by hikerguy** [comment]
Well, after trying what you said, there was no change. I tried a number of
things after that included uninstalling Brackets, going through Windows
registry and removing all traces of brackets, removing all remnants in
Explorer, rebooting, and still no change.

I then created a brand new file (which I should have done to start with
lol) and found that it worked in that file. I went back to my file and it
wouldn't work. I finally noticed that I had a </p> tag that was red
(because it didn't have a corresponding opening tag). After adding the
corresponding <p>, it started working again.

Is QM supposed to act this way?  I can maybe see it not preventing tags of
the same type being added until an issue is resolved, but why have it
disable all other tag types? I literally couldn't do any other type of QM
tag (h1, h2, div, etc). Is this something you might consider handling
another way?

Thanks for your help on this one.

Andy

**Comment by redmunds** [comment]
The Quick Markup code uses the CodeMirror editor tokenizer to determine what are start and end tags, so you document needs to have syntax that it considers valid. The red text seems like a nice hint to me. :)

You might also consider trying an HTML validation extension such as W3CValidator or one of the HTML hint/lint extensions.

**Comment by hikerguy** [comment]
Randy, I understand that. I was just wondering why it would prevent ANY tag
from being used when just the <p> tag had the issue. Is that just the way
it works (forcing you to fix it before you make any other mistakes)?

Agreed, the red was what caught my eye lol

**Comment by redmunds** [comment]
>  I was just wondering why it would prevent ANY tag from being used when just the <p> tag had the issue.

If you give me a concrete example I can take a closer look. It's possible that an improvement could be made.

**Comment by hikerguy** [comment]
Randy,

         Here's an example. Throw the following into Brackets, then
highlight the second paragraph (In iaculis, quam).  Next, try to use any
Quick Markup shortcut (e.g. Ctrl+h1, Ctrl+div, Ctrl+p). Nothing works.

Now, if you completely leave off the closing </p> tag, all shortcuts
(except Ctrl+p) will work.

When I'm editing an html doc, sometimes I get careless when moving things
around or deleting info, and it is possible to just have </> (with the
missing p). Maybe it's just me and I need to be more careful, but take a
look and see if you think it's worth the time to "correct".


<!DOCTYPE HTML>
<html lang="en">

<head>
<body>

 <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec lacinia
sed arcu ac efficitur. In et imperdiet nunc.</>

In iaculis, quam

non varius posuere, nisi elit ultricies nulla, id condimentum magna sem in
sapien. Pellentesque in feugiat tellus, at mollis libero. Integer vitae
accumsan nisl. Mauris blandit purus sed

purus iaculis, sit amet mattis lacus condimentum. Nulla quis pulvinar leo.
Etiam non sollicitudin erat, ut faucibus leo. Aliquam nisi tellus, dictum
id cursus ut, vulputate id erat. Nunc facilisis nec magna a varius.

</body>

</html>

**Comment by redmunds** [comment]
The Quick Markup extension is meant to be a tool that helps you markup text in a page that is in a good state. It's outside the scope of this tool to try to recover from invalid states.

I installed the W3CValidation extension and it pointed out the problem (and one other problem) immediately, so I recommend you use a tool intended for this sort of thing:

![w3c-validator](https://cloud.githubusercontent.com/assets/1197144/7359566/7fb754e8-ecf4-11e4-9ffe-b52374bec0c1.png)

Closing this issue.
